axon-server/administration/cluster-template.md

```diff
@@ -0,0 +1,84 @@
+{% endhint %}
+
+{% hint style="warn" %}
+As clustering is an Axon Server EE feature, license needs to be provided before using this feature.
+{% endhint %}
+
+
+## Usage
+
+To use Cluster template all you need to do is define a valid cluster template yml file. If this file is present on a clean Axon Server startup, it will automatically be picked up, and the cluster will be configured accordingly.
+
+{% hint style="warn" %}
+Each cluster node needs to have the cluster template yml file copy. Each node will read this file, find its own configuration and configure itself.
+{% endhint %}
+
+Default path from which Axon Server reads configuration is `./cluster-template.yml`
+
+You can override this path anytime by setting Axon Server property: 
+`axoniq.axonserver.clustertemplate.path:/mypath/cluster-template.yml`
+
+### Configuration
+
+Below you can find an example of basic cluster setup: _admin & default context in separate replication nodes, replicated across all nodes that are marked as primary.
+
+```yaml
+axoniq:
+  axonserver:
+    cluster-template:
+      first: axonserver-enterprise-1
+      replicationsGroups:
+      - name: _admin
+        roles:
+        - node: axonserver-enterprise-1
+          role: PRIMARY
+        - node: axonserver-enterprise-2
+          role: PRIMARY
+        - node: axonserver-enterprise-3
+          role: PRIMARY
+        contexts:
+        - name: _admin
+      - name: default
+        roles:
+        - node: axonserver-enterprise-2
+          role: PRIMARY
+        - node: axonserver-enterprise-3
+          role: PRIMARY
+        - node: axonserver-enterprise-1
+          role: PRIMARY
+        contexts:
+        - name: default
+      applications: []
+      users: []
+```
+
+![Cluster overview after default configuration is applied](/.gitbook/assets/cluster-template-default-configuration.png)
+_Cluster overview after default configuration is applied_
+
+### Export/Generator
+
+In order to avoid mistakes while writing with cluster configuration we implemented export button that will generate cluster template file based on current cluster configuration. 
+
+![Cluster Template export button location](/.gitbook/assets/cluster-template-export-button.png)
+_Location of export button at Settings page_
+
+**Recommended way for creating advanced cluster setup**
+
+* Start a fresh Axon Server (use basic cluster template setup mentioned above)
+* Configure a cluster via UI by creating users, applications, replication groups & contexts
+* Use export button located at Settings tab > Configuration panel  to download current cluster configuration
+* Replace basic cluster template with newly exported cluster template configuration
+
+
+{% hint style="warn" %}
+Use export button from main - leader node.
```

Comment:
```suggestion
Use the export button from the main / leader node.
```

Reply:
Would you mind elaborating why this is recommended? I have a hunch, but would be valuable for the readers too.

Reply:
One part is mentioned above:

_In order to avoid mistakes while writing a cluster configuration file, we have implemented an export button that will generate a cluster template file based on current setup._ 

But the real reason is:

Ideally, we would have cluster configuration generator on our website, that generates cluster-template.yml. Since we don't, right now, user needs somehow to create configuration before exporting it as cluster-template.yml to be used by the rest of the team, therefor are these recommended steps. It's easy to explain to you but I don't know how to write this down to ref guide... Any hints?

Alternative: skip this section and ask Dusan to create FE - cluster template generator. Would be a great feature for website

Reply:
Aah, what I mean is _why_ the export method should be called from a leader/main node specifically. It makes sense, but might be helpful for our readers.